### ISSUE TITLE
[LINT_BUG]: non-syntactic module names causes linter to fail

### ISSUE THREAD

**Issue by caldwellst** [Appsilon/box.linters#151]
### box.linters version

`0.10.5`

### Sample source code to lint

```r
box::use(./`__tests__`)
```

### Lint command used

```r
lintr::lint(
  code,
  linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)
```

### Lint result

```r
#> Linter 'box_unused_attached_mod_linter' failed in /lintr-test/a.R: <text>:1:4: unexpected input
#> 1: ./__
       ^
```

### Expected result

No error should have been generated. While non-syntactic names are rare to use, this is the recommended approach for [testing box modules](https://cran.rstudio.com/web/packages/box/vignettes/testing.html).
